### PR TITLE
Add linting for JSON and workflow to test on push

### PR DESCRIPTION
**Pull request by Johannes-Andersen**
There has been multiple issues around duplicate JSON enteries or broken JSON. So this should validate JSON issues.

Also added a GitHub action/workflow to show run status on pull requests and commits to make it easier to catch earlier. It it a free service for open source repos﻿

### CODE REVIEW

File: src/drivers/webextension/images/icons/Plataforma NEO.svg [hunk 1]
```diff
@@ -1,4 +0,0 @@
-<svg xmlns="http://www.w3.org/2000/svg" version="1.2" baseProfile="tiny" viewBox="0 0 51 51" overflow="scroll">
```
Comment (Johannes-Andersen) [inline src/drivers/webextension/images/icons/Plataforma NEO.svg:1]: This file was not used

File: src/drivers/webextension/_locales/ca/messages.json [hunk 2]
```diff
@@ -93,6 +93,6 @@
 	"categoryName71":           { "message": "Affiliate program" },
 	"categoryName72":           { "message": "Appointment scheduling" },
 	"categoryName73":           { "message": "Surveys" },
-	"categoryName75":           { "message": "A/B testing" },
+	"categoryName74":           { "message": "A/B testing" },
```
Comment (Johannes-Andersen) [inline src/drivers/webextension/_locales/ca/messages.json:96]: Example of what the new JSON lint caught

File: package.json [hunk 3]
```diff
@@ -8,15 +8,17 @@
...
+    "eslint-plugin-json": "^2.1.2",
     "eslint-plugin-nuxt": "^1.0.0",
     "eslint-plugin-prettier": "^3.1.4",
-    "prettier": "^2.1.0"
+    "prettier": "^2.1.2"
   },
   "scripts": {
     "link": "node ./bin/link.js",
-    "lint": "eslint --fix src/**/*.js",
+    "lint": "eslint src/**/*.{js,json}",
```
Comment (Johannes-Andersen) [inline package.json:20]: Moved to lint, and lint:fix. The runner should not fix linting, just throw the errors

File: package.json [hunk 4]
```diff
@@ -8,15 +8,17 @@
...
     "babel-eslint": "^10.1.0",
-    "eslint": "^7.7.0",
-    "eslint-config-prettier": "^6.11.0",
+    "eslint": "^7.13.0",
+    "eslint-config-prettier": "^6.15.0",
+    "eslint-plugin-json": "^2.1.2",
     "eslint-plugin-nuxt": "^1.0.0",
     "eslint-plugin-prettier": "^3.1.4",
-    "prettier": "^2.1.0"
+    "prettier": "^2.1.2"
```
Comment (Johannes-Andersen) [inline package.json:16]: Minor/Patch upgrades

File: bin/validate.js [hunk 5]
```diff
@@ -6,174 +6,167 @@ const { technologies, categories } = JSON.parse(
...
+
+  if (fs.statSync(filePath).isFile() && !file.startsWith('.')) {
+    if (!/^(png|svg)$/i.test(file.split('.').pop())) {
+      throw new Error(`Incorrect file type, expected PNG or SVG: ${filePath}`)
     }
-  })
-} catch (error) {
-  // eslint-disable-next-line no-console
-  console.error(error.message)
-}
```
Comment (Johannes-Andersen) [inline bin/validate.js:179]: Removed try/catch as we want the script to throw an error so we can fail the workflow.

**Comment by AliasIO** [comment]
Nice, thank you.

**Pull request merged**